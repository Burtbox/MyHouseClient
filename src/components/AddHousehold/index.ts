import AddHousehold, { AddHousehold as DisconnectedAddHousehold } from './AddHousehold';

export { DisconnectedAddHousehold };
export default AddHousehold;
