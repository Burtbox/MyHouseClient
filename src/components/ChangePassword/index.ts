import ChangePassword, { ChangePassword as DisconnectedChangePassword } from './ChangePassword';

export default ChangePassword;

export { DisconnectedChangePassword };
