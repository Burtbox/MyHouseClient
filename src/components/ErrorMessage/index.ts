import ErrorMessage, { ErrorMessage as DisconnectedErrorMessage } from './ErrorMessage';

export { DisconnectedErrorMessage };
export default ErrorMessage;
