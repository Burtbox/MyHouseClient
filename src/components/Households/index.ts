import Households, { Households as DisconnectedHouseholds } from './Households';

export default Households;

export { DisconnectedHouseholds };
