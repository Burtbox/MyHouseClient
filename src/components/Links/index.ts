import Links, { Links as DisconnectedLinks } from './Links';

export default Links;
export { DisconnectedLinks };
