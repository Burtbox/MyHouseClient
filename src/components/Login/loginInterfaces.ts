import { Action } from 'redux';
import { IComponentProps } from '../../interfaces/componentInterfaces';
import { IUserObject, IUserAuthenticationObject } from '../Users/usersInterfaces';
    
export interface ILoginProps extends IComponentProps {
    user: IUserObject;
    loggingIn: boolean;
}

export interface ILoginState {
    user: IUserAuthenticationObject;
    error: Error;
    loading: boolean;
}

export interface ILoginAction extends Action {
    loading: boolean;
}
