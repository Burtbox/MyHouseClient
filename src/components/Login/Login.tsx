import { Typography } from '@material-ui/core';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import * as React from 'react';
import { connect } from 'react-redux';
import { myHouseRoutes } from '../../enums/routesEnum';
import { IStore } from '../../interfaces/storeInterface';
import appStyles from '../../styles';
import Loading from '../Loading';
import { IUserAuthenticationObject } from '../Users/usersInterfaces';
import { loginUser } from './loginEpic'; // TODO: Remove this direct call?
import { ILoginProps, ILoginState } from './loginInterfaces';

export class Login extends React.Component<ILoginProps, ILoginState> {
    constructor(props: ILoginProps) {
        super(props);
        this.state = {
            user: {
                email: '',
                password: '',
            },
        };
    }

    handleLogin = (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault();
        const login: IUserAuthenticationObject = {
            email: this.state.user.email,
            password: this.state.user.password,
        };
        loginUser(this.props.dispatch, login);
    }

    handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        const target = event.target;
        const value = target.value;
        const name = target.name;

        this.setState(prevState => ({
            user: { ...this.state.user, [name]: value },
        }));
    }

    render() {
        return (
            <form style={appStyles.container} onSubmit={this.handleLogin}>
                <Typography variant="headline">Welcome</Typography>
                <div>
                    <TextField
                        name="email"
                        type="text"
                        label="Email Address"
                        placeholder="example@email.com"
                        required
                        onChange={this.handleInputChange}
                        disabled={this.props.loggingIn}
                        margin="normal"
                    />
                </div>
                <div>
                    <TextField
                        name="password"
                        type="password"
                        label="Password"
                        placeholder="**********"
                        autoComplete="current-password"
                        required
                        onChange={this.handleInputChange}
                        disabled={this.props.loggingIn}
                        margin="normal"
                    />
                </div>
                <div>
                    {this.props.loggingIn ? (
                        <Loading />
                    ) : (
                            <Button type="submit" variant="outlined">
                                Sign In
                            </Button>
                        )}
                </div>
                <div style={{ textAlign: 'center', marginTop: '3em' }}>
                    <Typography variant="subheading"> New to My House? </Typography>
                    <Button
                        variant="outlined"
                        color="secondary"
                        size="small"
                        onClick={() => this.props.history.push(myHouseRoutes.Register)}
                    >
                        Sign Up
                    </Button>
                </div>
            </form>
        );
    }
}

const mapStateToProps = (store: IStore) => {
    return {
        loggingIn: store.loginReducer.loggingIn,
    };
};

export default connect(mapStateToProps)(Login);
