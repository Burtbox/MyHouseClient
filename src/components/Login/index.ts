import Login, { Login as DisconnectedLogin } from './Login';

export default Login;

export { DisconnectedLogin };
