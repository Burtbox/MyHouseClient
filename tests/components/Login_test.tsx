import * as React from 'react';
import { shallow } from 'enzyme';
// import { shallow, mount, render } from 'enzyme';
import { DisconnectedLogin } from '../../src/components/Login';
import { ILoginProps } from '../../src/components/Login/loginInterfaces';

describe('Login test suite', () => {
    test('render a div', () => {
        const wrapper = shallow(
            <div>Hello Jest!</div>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('Render login screen', () => {
        const props: Readonly<ILoginProps> = {
            user: {
                userId: undefined,
                token: undefined,
                email: undefined,
                displayName: undefined,
            },
            history: undefined,
            loggingIn: false,
            dispatch: undefined,
        };
        const loginScreen = shallow(
            <DisconnectedLogin
                {...props}
            />,
        );
        expect(loginScreen).toMatchSnapshot();
    });
});
