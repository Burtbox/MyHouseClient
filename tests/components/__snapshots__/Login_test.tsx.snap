// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login test suite Render login screen 1`] = `
<form
  onSubmit={[Function]}
  style={
    Object {
      "marginTop": "20px",
      "textAlign": "center",
    }
  }
>
  <h2>
    Welcome
  </h2>
  <div>
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Email Address"
      fullWidth={false}
      hintText="example@email.com"
      maxlength="50"
      multiLine={false}
      name="email"
      onChange={[Function]}
      required={true}
      rows={1}
      type="text"
      underlineShow={true}
    />
  </div>
  <div>
    <TextField
      autoComplete="current-password"
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Password"
      fullWidth={false}
      hintText="**********"
      maxlength="30"
      multiLine={false}
      name="password"
      onChange={[Function]}
      required={true}
      rows={1}
      type="password"
      underlineShow={true}
    />
  </div>
  <div>
    <FlatButton
      disabled={false}
      fullWidth={false}
      label="Sign In"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      type="submit"
    />
  </div>
  <br />
  <div>
    <span
      style={
        Object {
          "vertical-align": "middle",
        }
      }
    >
       New to My House? 
    </span>
     
    <span>
      <FlatButton
        disabled={false}
        fullWidth={false}
        label="Sign Up"
        labelPosition="after"
        labelStyle={Object {}}
        onClick={[Function]}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        primary={false}
        secondary={true}
      />
    </span>
  </div>
</form>
`;

exports[`Login test suite render a div 1`] = `
<div>
  Hello Jest!
</div>
`;
